pkg/trace/writer/stats.go

```diff
@@ -95,6 +108,31 @@ func (w *StatsWriter) Run() {
 	}
 }
 
+// FlushSync is a no-op for TraceWriter
+func (w *StatsWriter) FlushSync() error {
+	if !w.flushSyncEnabled {
+		log.Error("SyncFlush called on StatsWriter, which is a no-op")
+	}
+
+	defer w.report()
+	w.wg.Add(1)
+	w.flushChan <- struct{}{}
+	w.wg.Wait()
+
+	// Wait for all the senders to finish
+	wg := sync.WaitGroup{}
+	for _, sender := range w.senders {
+		s := sender
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			s.waitForInflight()
+		}()
+	}
+	wg.Wait()
+	return nil
```

Comment:
I think we can remove the return value. This never returns anything else than `nil`.